vanilla-tests/src/main/java/fixtures/streamstyleserializationctorargs/models/Shark.java

```diff
@@ -0,0 +1,240 @@
+                                        + discriminatorValue
+                                        + "'.");
+                    }
+                });
+    }
+
+    static Shark fromJsonKnownDiscriminator(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(
+                reader -> {
+                    boolean lengthFound = false;
+                    float length = 0.0f;
+                    String species = null;
+                    List<Fish> siblings = null;
+                    boolean birthdayFound = false;
+                    OffsetDateTime birthday = null;
+                    Integer age = null;
+                    while (reader.nextToken() != JsonToken.END_OBJECT) {
+                        String fieldName = reader.getFieldName();
+                        reader.nextToken();
+
+                        if ("fishtype".equals(fieldName)) {
+                            String fishtype = reader.getString();
+                            if (!FISHTYPE.equals(fishtype)) {
+                                throw new IllegalStateException(
+                                        "'fishtype' was expected to be non-null and equal to '"
+                                                + FISHTYPE
+                                                + "'. The found 'fishtype' was '"
+                                                + fishtype
+                                                + "'.");
+                            }
+                        } else if ("length".equals(fieldName)) {
+                            length = reader.getFloat();
+                            lengthFound = true;
+                        } else if ("species".equals(fieldName)) {
+                            species = reader.getString();
+                        } else if ("siblings".equals(fieldName)) {
+                            siblings = reader.readArray(reader1 -> Fish.fromJson(reader1));
+                        } else if ("birthday".equals(fieldName)) {
+                            birthday =
+                                    reader.getNullable(
+                                            nonNullReader -> OffsetDateTime.parse(nonNullReader.getString()));
+                            birthdayFound = true;
+                        } else if ("age".equals(fieldName)) {
+                            age = reader.getNullable(JsonReader::getInt);
+                        } else {
+                            reader.skipChildren();
+                        }
+                    }
+                    if (lengthFound && birthdayFound) {
+                        Shark deserializedShark = new Shark(length, birthday);
+                        deserializedShark.setSpecies(species);
+                        deserializedShark.setSiblings(siblings);
+                        deserializedShark.age = age;
+
+                        return deserializedShark;
+                    }
```

Comment:
I assume changes are about this part?

Note that `--output-model-immutable` could make the ctor private. Though I think code will not be affected, as here it can call private ctor.

Reply:
The change being made is so that when there are no required constructor arguments deserialization will create an instance of the model eagerly and read JSON into the model instead of making references for each field and finally creating and populating the model.

This shouldn't have any issues with `--output-model-immutable` as that constructor would be accessible to the model itself, but I need to ask whether the constructor is accessible to sub-types.

Reply:
Seems to be `protected` https://github.com/Azure/autorest.java/blob/main/vanilla-tests/src/main/java/fixtures/bodycomplex/models/DotFish.java#L32

So should be good?

Reply:
Yeah, that should be good. Was less wondering for this PR specifically and more in general in the case models were spread across multiple packages.